afkak/brokerclient.py

```diff
@@ -144,47 +167,78 @@ def updateMetadata(self, new):
         self.host = new.host
         self.port = new.port
 
-    def makeRequest(self, requestId, request, expectResponse=True):
+    def makeRequest(self, correlationId, request, expectResponse=True):
         """
-        Send a request to our broker via our self.proto KafkaProtocol object.
+        Send a request to this broker.
+
+        The request may not be sent immediately, but it should be sent soon.
+        A connection to the broker is created when the first request is made.
+
+        Each request is sent at most once per broker connection. If the
+        connection drops while requests are outstanding requests are resent
+        upon reconnection in the order originally issued.
+
+        This method returns a deferred that fires with the `bytes` of the
+        response is from the broker. It may fail in a few ways:
 
-        Return a deferred which will fire when the reply matching the requestId
-        comes back from the server, or, if expectResponse is False, then
-        return None instead.
-        If we are not currently connected, then we buffer the request to send
-        when the connection comes back up.
+        - With `ClientError` when the `_KafkaBrokerClient` is closed.
+        - With `twisted.internet.defer.CancelledError` if its :meth:`cancel()
+          <twisted.internet.defer.Deferred.cancel>`_ method is called.
+        - With some other exception in the case of bugs.
+
+        Cancelling the deferred _may_ prevent the request from being sent to
+        a broker.
+
+        :param int corelationId:
+            ID number used to match responses with requests. This must match
+            the number embedded in *request*.
+
+        :param bytes request:
+            The serialized request PDU (not including the length prefix).
+
+        :parma bool expectResponse:
+            Will the Kafka broker send a response? Unsetting this flag changes
+            the result:
```

Comment:
Why is this useful? Just asking for context.

Reply:
When you produce a message with acks=0 the broker doesn't send a response at all, so if we kept around an entry in the requests table for it we'd effectively leak memory (and resent the request forever).